bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationServiceOSGiTest.java

```diff
@@ -0,0 +1,290 @@
+                "@text/offline.with-two-params [@text/channel-type.magic.alert.label, 60]"));
+        ThingStatusInfo info = thingStatusInfoI18nLocalizationService.getLocalizedThingStatusInfo(thing, Locale.GERMAN);
+        assertThat(info, is(new ThingStatusInfo(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                "Thing ist wegen Alarm offline. Bitte versuchen Sie es in 60 Sekunden erneut.")));
+    }
+
+    @Before
+    public void setup() throws IOException {
+        LocaleProvider localeProvider = getService(LocaleProvider.class);
+        assertThat(localeProvider, is(notNullValue()));
+        defaultLocale = localeProvider.getLocale();
+
+        new DefaultLocaleSetter(getService(ConfigurationAdmin.class)).setDefaultLocale(Locale.ENGLISH);
+        waitForAssert(() -> assertThat(localeProvider.getLocale(), is(Locale.ENGLISH)));
+
+        registerVolatileStorageService();
+
+        SimpleThingHandlerFactory simpleThingHandlerFactory = new SimpleThingHandlerFactory();
+        ComponentContext componentContext = mock(ComponentContext.class);
+        when(componentContext.getBundleContext()).thenReturn(bundleContext);
+        simpleThingHandlerFactory.activate(componentContext);
+        registerService(simpleThingHandlerFactory, ThingHandlerFactory.class.getName());
+
+        thing = ThingBuilder.create(new ThingTypeUID("aaa:bbb"), "ccc").build();
+
+        managedThingProvider = getService(ManagedThingProvider.class);
+        assertThat(managedThingProvider, is(notNullValue()));
+
+        managedThingProvider.add(thing);
+
+        waitForAssert(() -> assertThat(thing.getStatus(), is(ThingStatus.ONLINE)));
+
+        thingStatusInfoI18nLocalizationService = getService(ThingStatusInfoI18nLocalizationService.class);
+        assertThat(thingStatusInfoI18nLocalizationService, is(notNullValue()));
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        managedThingProvider.remove(thing.getUID());
+        new DefaultLocaleSetter(getService(ConfigurationAdmin.class)).setDefaultLocale(defaultLocale);
+        waitForAssert(() -> assertThat(getService(LocaleProvider.class).getLocale(), is(defaultLocale)));
+    }
+
+    @SuppressWarnings("null")
+    private void setThingStatusInfo(Thing thing, ThingStatusInfo thingStatusInfo) {
+        ((SimpleThingHandler) thing.getHandler()).setThingStatusInfo(thingStatusInfo);
+    }
+
+    private class SimpleThingHandlerFactory extends BaseThingHandlerFactory {
+
+        @Override
+        public void activate(ComponentContext componentContext) {
+            super.activate(componentContext);
+        }
+
+        @Override
+        public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+            return true;
+        }
+
+        @Override
+        protected ThingHandler createHandler(Thing thing) {
+            return new SimpleThingHandler(thing);
+        }
+    }
+
+    private class SimpleThingHandler extends MagicColorLightHandler {
```

Comment:
Is there any advantage from extending the magic handler? I'm not sure if we should start depending our unit tests on the magic bundle. What about extending the BaseThingHandler?